WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.java

```diff
@@ -0,0 +1,395 @@
+import org.wordpress.android.fluxc.store.PluginStore;
+import org.wordpress.android.ui.plugins.PluginBrowserActivity;
+import org.wordpress.android.ui.plugins.PluginUtils;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class PluginBrowserViewModel extends AndroidViewModel {
+    public enum PluginListStatus {
+        CAN_LOAD_MORE,
+        DONE,
+        ERROR,
+        FETCHING,
+        LOADING_MORE
+    }
+
+    @Inject Dispatcher mDispatcher;
+    @Inject PluginStore mPluginStore;
+
+    private String mSearchQuery;
+    private SiteModel mSite;
+
+    private final Handler mHandler;
+
+    private MutableLiveData<PluginListStatus> mNewPluginsListStatus;
+    private MutableLiveData<PluginListStatus> mPopularPluginsListStatus;
+    private MutableLiveData<PluginListStatus> mSitePluginsListStatus;
+    private MutableLiveData<PluginListStatus> mSearchPluginsListStatus;
+
+    private MutableLiveData<List<WPOrgPluginModel>> mNewPlugins;
+    private MutableLiveData<List<WPOrgPluginModel>> mPopularPlugins;
+    private MutableLiveData<List<SitePluginModel>> mSitePlugins;
+    private MutableLiveData<List<WPOrgPluginModel>> mSearchResults;
+
+    private MutableLiveData<String> mLastUpdatedWpOrgPluginSlug;
+
+    public PluginBrowserViewModel(@NonNull Application application) {
+        super(application);
+
+        ((WordPress) application).component().inject(this);
+        mDispatcher.register(this);
+
+        mHandler = new Handler();
+
+        mSitePlugins = new MutableLiveData<>();
+        mNewPlugins = new MutableLiveData<>();
+        mPopularPlugins = new MutableLiveData<>();
+        mSearchResults = new MutableLiveData<>();
+
+        mNewPluginsListStatus = new MutableLiveData<>();
+        mPopularPluginsListStatus = new MutableLiveData<>();
+        mSitePluginsListStatus = new MutableLiveData<>();
+        mSearchPluginsListStatus = new MutableLiveData<>();
+        mLastUpdatedWpOrgPluginSlug = new MutableLiveData<>();
+    }
+
+    @Override
+    protected void onCleared() {
+        super.onCleared();
+        mDispatcher.unregister(this);
+    }
+
+    public void start() {
+        reloadAllPluginsFromStore();
+
+        fetchPlugins(PluginBrowserActivity.PluginListType.SITE, false);
+        fetchPlugins(PluginBrowserActivity.PluginListType.POPULAR, false);
+        fetchPlugins(PluginBrowserActivity.PluginListType.NEW, false);
+    }
+
+    // Site & WPOrg plugin management
+
+    public WPOrgPluginModel getWPOrgPluginForSitePlugin(SitePluginModel sitePlugin) {
+        if (sitePlugin == null) {
+            return null;
+        }
+        return PluginUtils.getWPOrgPlugin(mPluginStore, sitePlugin);
+    }
+
+    public SitePluginModel getSitePluginFromSlug(String slug) {
+        return mPluginStore.getSitePluginBySlug(getSite(), slug);
```

Comment:
I did not see this comment when I wrote [this](https://github.com/wordpress-mobile/WordPress-Android/pull/7229#issuecomment-365719942), it was just a coincidence. I added your suggestion in 0aac57a2df5bc18679f9ac9fe50f750851f14af7 with a `TODO` to remove it if we add caching to `PluginStore`. As you said, if we still have the information, there is no reason not to use it here. It doesn't introduce another `HashMap` for us to juggle, so a nice middle ground. Having said that, if I can find time for it, I do want to look into how we can cache the information in FluxC because I think this feature can greatly benefit from it.